java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -208,6 +235,14 @@ public void visitVariable(VariableTree tree) {
     }
   }
 
+  @Override
+  public void visitWhileStatement(WhileStatementTree tree) {
+    ConditionalState conditionalState = visitCondition(tree.condition());
+    currentState = conditionalState.trueState;
+    scan(tree.statement());
+    currentState = currentState.parentState.merge(conditionalState.trueState, conditionalState.falseState);
```

Comment:
if a=null before the loop, and a!=null in the condition, then the merge will set the value of a to UNKNOWN after the loop, which is valuable. I'll add a comment in that sense.